packages/api-sdk/src/resources/version.ts

```diff
@@ -111,6 +111,24 @@ class VersionResource extends CrudResource<BaseModels.Version.Model<BaseModels.V
     return data;
   }
 
+  public async getDiagramsByIDs<T extends Partial<BaseModels.Diagram.Model>>(versionId: string, diagramIds: string[]): Promise<T[]>;
+
+  public async getDiagramsByIDs<T extends BaseModels.BaseDiagramNode = BaseModels.BaseDiagramNode>(
+    versionId: string,
+    diagramIds: string[]
+  ): Promise<BaseModels.Diagram.Model<T>[]>;
+
+  public async getDiagramsByIDs<T extends BaseModels.Diagram.Model<any> = BaseModels.Diagram.Model>(
+    versionId: string,
+    diagramIds: string[]
+  ): Promise<T[]>;
+
+  public async getDiagramsByIDs(versionId: string, diagramIds: string[]): Promise<BaseModels.Diagram.Model[]> {
+    const { data } = await this.fetch.get<BaseModels.Diagram.Model[]>(`${this._getCRUDEndpoint(versionId)}/diagrams${this._getIDsQuery(diagramIds)}`);
```

Comment:
The [`URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL) class is really helpful for building urls. Instead of tying to build it manually (especially query params!)

```ts
const url = new URL('diagrams', this._getCRUDEndpoint(versionId));
url.searchParams.set('ids', diagramIds.join(','));

const { data } = await this.fetch.get<BaseModels.Diagram.Model[]>(url.href);
```

Reply:
Express uses the `qs` package by default to parse these query parameters. That library supports [array queries](https://www.npmjs.com/package/qs#user-content-parsing-arrays) using the `id[]` syntax.

That means no `.split(',')` on the server side, and instead the query param will already be an array.

```ts
const url = new URL('diagrams', this._getCRUDEndpoint(versionId));
diagramIds.forEach((id) => url.searchParams.append('ids[]', id));
```

Reply:
most of the sdk uses this string joining method, should i open a ticket to change this in a future PR?

Reply:
I think for this endpoint we should use the alternative array loading format: https://masteringjs.io/tutorials/express/query-parameters